test/e2e/bgppolicy_test.go

```diff
@@ -0,0 +1,299 @@
+	require.Equal(t, 0, rc)
+
+	_, err = data.updateServiceInternalTrafficPolicy("agnhost-svc", true)
+	require.NoError(t, err)
+
+	t.Log("Update the test BGPPolicy with a new local ASN")
+	updatedBGPPolicy := bgpPolicy.DeepCopy()
+	updatedBGPPolicy.Spec.LocalASN = updatedLocalASN
+	updatedBGPPolicy.Spec.Advertisements.Pod = nil
+	_, err = data.crdClient.CrdV1alpha1().BGPPolicies().Update(context.TODO(), updatedBGPPolicy, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Log("Get routes installed on remote FRR router and verify them")
+	expectedRoutes = []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: []string{nodeIPv4(0)}}}
+	notExpectedRoutes := []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()}}
+	checkFRRRouterBGPRoutes(t, routesToStrings(expectedRoutes), routesToStrings(notExpectedRoutes))
+
+	t.Log("verify the connectivity of the installed routes on remote FRR route")
+	ipsToConnect = []string{clusterIP}
+	for _, ip := range ipsToConnect {
+		cmd := fmt.Sprintf("/usr/bin/wget -O - http://%s:8080/hostname -T 5", ip)
+		rc, stdout, _, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, cmd, "/", nil, "")
+		require.NoError(t, err)
+		require.Equal(t, 0, rc)
+		require.Equal(t, podName, stdout)
+	}
+}
+
+func checkFRRRouterBGPRoutes(t *testing.T, expectedRouteStrings, notExpectedRouteStrings []string) {
+	t.Helper()
+	err := wait.PollUntilContextTimeout(context.Background(), time.Second, 30*time.Second, true, func(context.Context) (bool, error) {
+		gotRoutes, err := dumpFRRRouterBGPRoutes()
+		if err != nil {
+			return false, err
+		}
+		gotRouteStrings := routesToStrings(gotRoutes)
+		gotRoutesSet := sets.NewString(gotRouteStrings...)
+		notExpectedRoutesSet := sets.NewString(notExpectedRouteStrings...)
+		expectedRoutesSet := sets.NewString(expectedRouteStrings...)
+		if !gotRoutesSet.IsSuperset(expectedRoutesSet) {
+			return false, nil
+		}
+		if notExpectedRoutesSet.Len() != 0 && gotRoutesSet.IsSuperset(notExpectedRoutesSet) {
+			return false, nil
+		}
+		return true, nil
+	})
+
+	if err != nil {
+		t.Fatalf("Failed to get the expected BGP routes: %v", err)
+	}
+}
+
+func runVtyshCommands(commands []string) (int, string, string, error) {
+	return exec.RunDockerExecCommand(externalInfo.externalFRRCID, "/usr/bin/vtysh", "/", nil, strings.Join(commands, "\n"))
+}
+
+func generateVtyshCommandsForConfiguringBGP(localASN, remoteASN int32) []string {
+	commands := []string{
+		"configure terminal",
+		fmt.Sprintf("router bgp %d", localASN),
+		"no bgp ebgp-requires-policy",
+		"no bgp network import-check",
+	}
+	for _, node := range clusterInfo.nodes {
+		commands = append(commands, fmt.Sprintf("neighbor %s remote-as %d", node.ipv4Addr, remoteASN))
+		commands = append(commands, fmt.Sprintf("neighbor %s password %s", node.ipv4Addr, bgpPeerPassword))
+	}
+	commands = append(commands,
+		"exit",
+		"exit",
+		"write memory")
+	return commands
+}
+
+func generateVtyshCommandsForClearingBGP(localASN int32) []string {
```

Comment:
not sure why you didn't define `configureBGP` and `cleanupBGP` instead? (both function would ultimately call `runVtyshCommands`).